checker/src/abstract_value.rs

```diff
@@ -2691,6 +2729,60 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         self.get_cached_interval()
     }
 
+    #[logfn_inputs(TRACE)]
+    fn get_cached_octagon(&self) -> Rc<OctagonDomain> {
+        {
+            let mut cached_octagon = self.octagon.borrow_mut();
+            let octagon_opt = cached_octagon.as_ref();
+            if let Some(octagon) = octagon_opt {
+                return octagon.clone();
+            }
+            let octagon = self.get_as_octagon();
+            *cached_octagon = Some(Rc::new(octagon));
+        }
+        self.get_cached_octagon()
+    }
+
+    fn get_as_octagon(&self) -> OctagonDomain {
+        if self.expression_size > k_limits::MAX_EXPRESSION_SIZE / 10 {
+            return octagon_domain::BOTTOM;
+        }
+        match &self.expression {
+            Expression::Top => octagon_domain::BOTTOM,
+            Expression::CompileTimeConstant(ConstantDomain::I128(val)) => (*val).into(),
+            //Expression::CompileTimeConstant(ConstantDomain::U128(val)) => (*val).into(),
```

Comment:
If val fits into an I128 you could still produce a domain value for this expression.